### PR TITLE
libreoffice-fresh: strip away BUILDCONFIG, reduce runtime closure size by ~20%

### PR DESCRIPTION
**Pull request by Ma27**
###### Description of changes
In v7.5.x a change was introduced that writes the BUILDCONFIG into `$out/lib/libreoffice/program/libsofficeapp.so` including the `PKG_CONFIG_PATH` containing references to all `dev` outputs of library dependencies:

    $ strings $(nix-build -A libreoffice-fresh)/lib/libreoffice/program/libsofficeapp.so|grep PKG_CONFIG_PATH
    [...], "BuildConfig": "[...] 'PKG_CONFIG_PATH=[...]'"

This isn't really needed because this information can also be obtained by `nix derivation show`. Also, this causes a 20% larger runtime-closure because of all the dev dependencies being referenced by the output and thus downloaded whenever libreoffice is substituted somewhere. The actual numbers look like this:

    $ nix path-info -Sh ./result-old
    /nix/store/3mzrqh4gg7v27vdrrap9dj3x8myysmyf-libreoffice-7.5.4.1-wrapped	  2.0G
    $ nix path-info -Sh ./result
    /nix/store/g5y60s0a2q2v6r58xcayv62z7fjfi816-libreoffice-7.5.4.1-wrapped	  1.6G

Only `libreoffice-fresh` is affected, `pkgs.libreoffice` isn't because it still points to 7.4 whereas the problematic change was introduced in 7.5.

To make sure this doesn't get reintroduced by accident, the derivation also prohibits now to reference any dev output from a build input.

[1] https://gerrit.libreoffice.org/c/core/+/141197

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by doronbehar — APPROVED** [review]
Looks great to me! Thanks for noticing and making the effort. Very much appreciated :rocket:.

File: pkgs/applications/office/libreoffice/0001-Strip-away-BUILDCONFIG.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+diff --git a/desktop/source/lib/init.cxx b/desktop/source/lib/init.cxx
+index 8d830c0cbd00..289d10ddbe10 100644
+--- a/desktop/source/lib/init.cxx
++++ b/desktop/source/lib/init.cxx
+@@ -7097,7 +7097,7 @@ static char* lo_getVersionInfo(SAL_UNUSED_PARAMETER LibreOfficeKit* /*pThis*/)
+         "\"ProductVersion\": \"%PRODUCTVERSION\", "
+         "\"ProductExtension\": \"%PRODUCTEXTENSION\", "
+         "\"BuildId\": \"%BUILDID\", "
+-        "\"BuildConfig\": \""  BUILDCONFIG  "\" "
++        "\"BuildConfig\": \""  "\" "
```
Comment (NickCao) [inline pkgs/applications/office/libreoffice/0001-Strip-away-BUILDCONFIG.patch:26]: Can we add some informational text here for the wondering.

**Review by 7c6f434c — APPROVED** [review]

**Comment by Ma27** [comment]
@NickCao done!

**Comment by K900** [comment]
Pretty sure this broke `libreoffice-qt`.

**Comment by Ma27** [comment]
can take a look later.

**Comment by doronbehar** [comment]
> Pretty sure this broke `libreoffice-qt`.

Indeed! It happens in the end of it unfortunately - many of the `dev` outputs are referenced and Nix fails the build:

```
error: output '/nix/store/2y0czyy26gcsqhmcvd2mlqa35f0gcl4l-libreoffice-7.5.4.1' is not allowed to refer to the following paths:
         /nix/store/0hmvklj0mbhrn8flwbcwivvkv45limhg-freetype-2.13.0-dev
         /nix/store/0rnx7rc87hwkbrhsys7mgwq4jw6pz7ma-zlib-1.2.13-dev
         /nix/store/13p48hi1bfgpwgbqqx3s6c0i126gvv96-libXinerama-1.1.5-dev
         /nix/store/1b62rr75sfz00qixad3s4s2zj0c3g8q9-libXext-1.3.5-dev
         /nix/store/2a34x9kp91dlsbg06w903cpv3r3ihxsf-kio-5.108.0-dev
         /nix/store/2xm5rq23mqbsz00g719qymq6fcnpz3cr-gtk+3-3.24.38-dev
         /nix/store/6qqnfjv51qf2xpasssjnsk55sq7072ij-libjpeg-turbo-2.1.5.1-dev
         /nix/store/7m1px7m2zvidm7w3s93qjgvpkvdzwp8c-qtbase-5.15.9-dev
         /nix/store/7y7plvhsnpb841zd0fwk2q5sp9ah2n6v-libGL-1.6.0-dev
         /nix/store/82p8xgcrb9qn01rabzh6a200yvh4kqhk-cups-2.4.6-dev
         /nix/store/99hw785dmzxllzjxi89palmmd81n1g5c-kcoreaddons-5.108.0-dev
         /nix/store/b8dz2wfxj33ivrq39ga8h0ails3qv3li-graphite2-1.3.14-dev
         /nix/store/db79xky20dvsvs5546ffpv5db70yjlxq-icu4c-73.2-dev
         /nix/store/dcwr090bqdp45z21cdzx6pngacch97mb-glib-2.76.4-dev
         /nix/store/iwnn064wbl64f9vq5gh35y06grr55633-libX11-1.8.6-dev
         /nix/store/k1rfz8y5bgnwkr0y483k4mpy7k9v8p13-libxml2-2.11.4-dev
         /nix/store/l0vnx3ik63qrlj086gq06fjmalsi5dpy-openssl-3.0.9-dev
         /nix/store/mkx285y7r04x83ya9fgrh4wrfdhqn10g-libxslt-1.1.38-dev
         /nix/store/pl74gzzdjagqybwg4srwgvg233rxg12d-expat-2.5.0-dev
         /nix/store/s8filxypfn1c32vcvq6mzxyds00cp52n-libXi-1.8.1-dev
         /nix/store/sfw83wf115jb435rbqql9apmvd2l4y2i-cairo-1.16.0-dev
         /nix/store/wyrba0f8h73cb7ik64myqphi7217fz0g-fontconfig-2.14.2-dev
         /nix/store/x2d20gpjfvjk6raf6ycyskvqh958fj5b-qtx11extras-5.15.9-dev
error: 1 dependencies of derivation '/nix/store/ki2pw9h0638zfijbfsrb40v6nf2qb2ij-libreoffice-7.5.4.1-wrapped.drv' failed to build
```

**Comment by Ma27** [comment]
Yeah, because of the QT variant a few more dev outputs are needed already, but that isn't something I'll investigate. Instead, I'd just skip the `disallowedRequisites` for `libreoffice-qt` and keep it for the non-QT variants to make sure we don't miss a new buildconfig-like closure bloat in the future.

Already prepared a patch locally, currently building all variants, a PR will follow...

**Comment by Ma27** [comment]
@k900 #245518 %)

**Pull request merged**